hazelcast/src/main/resources/hazelcast-full-example.xml

```diff
@@ -2540,6 +2574,316 @@
+            * enabled:
+               Specifies whether the member address provider SPI is enabled or not. Its default value is false.
+            It has the following sub-elements:
+                - <class-name>:
+                    The name of the class implementing the com.hazelcast.spi.MemberAddressProvider interface.
+                - <properties>:
+                    The properties that will be provided when constructing the provided MemberAddressProvider. Hazelcast will
+                    first try instantiating the provided class by invoking a constructor accepting a single
+                    java.util.Properties instance. In the case where there is no such constructor and there are also
+                    no properties defined by this configuration, Hazelcast will exceptionally try to use the no-arg
+                    constructor.
+        * <failure-detector>:
+            A failure detector is responsible to determine if a member in the cluster is unreachable or crashed.
+            Please refer to https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#failure-detector-configuration
+            for the failure detectors implemented by Hazelcast.
+             
+            This element has the following sub-element:
+            * <icmp>:
+                ICMP can be used in addition to the other detectors. It operates at layer 3 detects network
+                and hardware issues more quickly.
+
+                It has the following sub-elements:
+                * <timeout-milliseconds>:
+                    Timeout in Milliseconds before declaring a failed ping.
+                * <fail-fast-on-startup>:
+                    Cluster Member will fail to start if it is unable to action an ICMP ping command when ICMP is enabled.
+                    Failure is usually due to OS level restrictions.
+                * <interval-milliseconds>:
+                    Time in milliseconds between each ICMP ping.
+                * <max-attempts>:
+                    Maximum number of consecutive failed attempts before declaring a member suspect.
+                * <parallel-mode>:
+                    Run ICMP detection in parallel with the Heartbeat failure detector.
+                * <ttl>:
+                    Maximum number of times the IP Datagram (ping) can be forwarded, in most cases
+                    all Hazelcast cluster members would be within one network switch/router therefore
+                    default of 0 is usually sufficient.
+
+        * <member-server-socket-endpoint-config>:
+            Configures the server socket used for the member-member communication.
+        * <client-server-socket-endpoint-config>:
+            Configures the server socket used for the member-client communication.
+        * <wan-server-socket-endpoint-config>:
+            Configures the server socket used for the WAN replication.
```

Comment:
Maybe worth to indicate here that more than one WAN server socket can be configured?

Reply:
Done.

Reply:
This comment revealed https://github.com/hazelcast/hazelcast/issues/14800. On it.